docs/language_guide.md

```diff
@@ -0,0 +1,619 @@
+
+### Application, Endpoint
+As described above, here is how you can start to define an application.
+
+`mobile.sysl`
+```
+MobileApp:
+  Login: ...
+  Search: ...
+  Order: ...
+```
+
+Sysl expects all lines after `:` to be indented. The only exception to this rule is when you want to use the shortcut `...`. Sysl has support to generate code for defining `Rest` api servers. See [Rest](#rest) for more details.
+
+### Data Models
+#### Data Types
+Sysl supports following data types out of the box.
+  * int, int64, int32
+  * float, decimal
+  * string
+  * bool
+  * datetime, date
+  * any
+  * xml
+
+#### Relational Data Model
+Relational Data model is the most common way of persisting data in a database. You can define your data model directly in sysl.
+`data.sysl`
+```java
+CustomerOrderModel:
+  !table Customer:
+    customer_id <: int [~pk]
+
+  !table CustomerOrder:
+    order_id <: int [~pk]
+    customer_id <: Customer.customer_id
+```
+In the above example:
+  *  Customer.customer_id is a primary key.
+  *  CustomerOrder has a foreign key customer_id which refers to the primary key of Customer (i.e. customer_id).
+
+See [data.sysl](assets/data.sysl) for complete example for Relation model.
+
+![](assets/data-Relational-Model.png)
+
+#### Object Model
+Define a typical in memory Object model of an application
+```java
+CustomerOrderModel:
+  !type Address:
+    line_1 <: string
+    city <: string
+
+  !type Customer:
+    customer_id <: int
+    addresses <: set of Address
+
+  !type CustomerOrder:
+    order_id <: int
+    customer <: Customer
+```
+Note:
+
+  * `set of Address` - Set is the only collection type
+
+See [data.sysl](assets/data.sysl) for complete example for Object model.
+
+![](assets/data-Object-Model.png)
+
+#### Generate data model diagrams using the following command
+```
+sysl data --project Relational -o relational.png data
+sysl data --project ObjectModel -o object.png data
+```
+#### Generate code using the following command
+```
+reljam model data Relational
+reljam model data ObjectModel
+```
+where
+  * `reljam`- Relational Java Model exporter
+  * `model` - generate "model" code
+  * `data`  - input data.sysl file.
+  * `Relational` or `ObjectModel` - name of the Data model to generate code for.
+
+You should see a Java file generated for each type. See [Data Models](#data-model) for more details.
+
+##### Nested Data Models
+
+Sysl has a shorter syntax to define types as well.
+```java
+Model:
+  !type Request:
+    header <:
+      header_id <: int
+    body <:
+      details <: string
+```
+At the moment, reljam does not generate code for this. See issue [#155](https://github.com/anz-bank/sysl/issues/155).
+
+### Imports
+
+To keep things modular, sysl allows you to import definitions created in other `.sysl` files.
+
+E.g. `server.sysl`
+```
+Server:
+  Login: ...
+  Register: ...
+```
+
+and you use `import` in `client.sysl`
+
+```js
+import server
+
+Client:
+  Login:
+    Server <- Login
```

Comment:
sysl-language complaint -- i'm offended that symbols get imported into the global namespace.

Now I have to exhaustively read all files that are (transitively?) imported to find where `Server` is defined. or search across the project for `Server:`, i guess. or perhaps use some smart sysl language IDE plugin which can take me to the definition.